### PR TITLE
DT-2493 Allow user to amend a role name

### PR DESCRIPTION
**Pull request by karenmillermoj**

### CODE REVIEW

**Review by petergphillips — DISMISSED** [review]
LGTM

File: src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/resource/api/RolesController.kt [hunk 1]
```diff
@@ -151,3 +182,10 @@ data class RoleAdminType(
     AdminType.valueOf(a.uppercase(getDefault())).description
   )
 }
+
+@ApiModel(description = "Role Name")
+data class RoleAmendment(
+  @ApiModelProperty(required = true, value = "Role Name", example = "Central admin")
+  @field:NotBlank(message = "Role name must be supplied")
```
Comment (petergphillips) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/resource/api/RolesController.kt:189]: should really validate the min and max length here too (between 4 and 100?)

Comment (karenmillermoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/resource/api/RolesController.kt:189]: validation is in the ui side but can add here too

**Review by SimonMitchellMOJ — APPROVED** [review]
LGTM

**Pull request merged**